docs/swift/DesignDoc.md

```diff
@@ -95,13 +117,13 @@ unique type symbol emitted in the generated binary.
 
 Users are likely to balk at the ugliness of underscore-delimited names for every
 generated type. To improve upon this situation, we will add a new string file
-level option, `swift_package_typealias`, that can be added to `.proto` files.
-When present, this will cause `typealias`es to be added to the generated Swift
-messages that replace the package name prefix with the provided string. For
-example, the following `.proto` file:
+level option, `swift_package_prefix`, that can be added to `.proto` files.
+When present, this will cause the package portion of the top-level type names
+to be replaced with the provided string. For example, the following `.proto`
+file:
 
 ```protobuf
-option swift_package_typealias = "FBP";
+option swift_package_prefix = "FBP";
```

Comment:
Updated with `swift_proto_package_prefix`, since I prefer being more explicit to less explicit, and `swift_prefix` feels a bit too general.